lib/concurrent/mvar.rb

```diff
@@ -68,10 +68,14 @@ def take(timeout = nil)
 
         # If we timed out we'll still be empty
         if unlocked_full?
-          value = @value
-          @value = EMPTY
-          @empty_condition.signal
-          apply_deref_options(value)
+          if block_given?
+            yield @value
```

Comment:
Why would the item not be removed from the MVar when a block is given? That seems semantically inconsistent.